### PR TITLE
Mes-6867: Migrate e2e practice mode into des4

### PR DESCRIPTION
**Pull request by andrewsetterfield**
## Description

Migrate e2e practice mode into des4

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

## Screenshots (optional
Screenshot 2021-11-17 at 15 45 25
Screenshot 2021-11-17 at 15 45 38
Screenshot 2021-11-17 at 15 45 44
Screenshot 2021-11-17 at 15 46 01
Screenshot 2021-11-17 at 15 46 35
Screenshot 2021-11-17 at 15 46 10
)

### CODE REVIEW

File: src/app/pages/test-report/components/toolbar/toolbar.scss [hunk 1]
```diff
@@ -1,14 +1,16 @@
 :host {
-  #wrapper {
-    height: 68px;
-    padding: 12px 24px;
+  .toolbar-wrapper {
+    height: 56px;
+    padding: 6px 24px;
     background: var(--mes-white);
+    //padding-top: 6px;
+    //padding-bottom: 4px;
```
Comment (matthew2564) [inline src/app/pages/test-report/components/toolbar/toolbar.scss]: please remove

File: src/app/pages/test-report/components/toolbar/toolbar.scss [hunk 2]
```diff
@@ -1,14 +1,16 @@
 :host {
-  #wrapper {
-    height: 68px;
-    padding: 12px 24px;
+  .toolbar-wrapper {
+    height: 56px;
+    padding: 6px 24px;
     background: var(--mes-white);
+    //padding-top: 6px;
```
Comment (andrewsetterfield) [inline src/app/pages/test-report/components/toolbar/toolbar.scss]: will remove

File: src/store/tests/rekey-reason/rekey-reason.actions.ts [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
-  '[RekeyReasonPage] Rekey Reason Other Selected',
+  '[FakeJournalPage] Rekey Reason Other Selected',
   (otherSelected: boolean) => ({ otherSelected }),
 );
 
 export const OtherReasonUpdated = createAction(
-  '[RekeyReasonPage] Rekey Reason Other Reason Updated',
+  '[FakeJournalPage] Rekey Reason Other Reason Updated',
   (reason: string) => ({ reason }),
 );
```
Comment (matthew2564) [inline src/store/tests/rekey-reason/rekey-reason.actions.ts]: Can we undo this entire file please?

File: src/components/test-slot/test-outcome/test-outcome.ts [hunk 4]
```diff
@@ -183,8 +183,10 @@ export class TestOutcomeComponent implements OnInit {
   }
 
   async startTest() {
+    console.log('isE2EPracticeMode', this.isE2EPracticeMode());
     if (this.isE2EPracticeMode()) {
-      // this.store$.dispatch(StartE2EPracticeTest(this.slotDetail.slotId.toString()));
+      console.log('hello there');
```
Comment (matthew2564) [inline src/components/test-slot/test-outcome/test-outcome.ts]: Please remove log here and on line 186

File: src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts [hunk 5]
```diff
@@ -217,7 +220,19 @@ export abstract class WaitingRoomToCarBasePageComponent extends BasePageComponen
   }
 
   async onViewTestCentreJournal(): Promise<void> {
-    await this.router.navigate([TEST_CENTRE_JOURNAL_PAGE]);
+    if (this.isEndToEndPracticeMode) {
+      await this.presentAlert();
+    } else await this.router.navigate([TEST_CENTRE_JOURNAL_PAGE]);
+  }
```
Comment (matthew2564) [inline src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts]: Pedantic i guess, but DES code often favours returning and then removing the else i.e.
```
async onViewTestCentreJournal(): Promise<void> { 
   if (this.isEndToEndPracticeMode) {
        await this.presentAlert();
        return;
   }
  await this.router.navigate([TEST_CENTRE_JOURNAL_PAGE]);
}
```

File: src/components/test-slot/test-outcome/test-outcome.ts (hunk 4 above)
Comment (andrewsetterfield) [reply src/components/test-slot/test-outcome/test-outcome.ts]: there's always one.  removing

File: src/store/tests/rekey-reason/rekey-reason.actions.ts (hunk 3 above)
Comment (andrewsetterfield) [reply src/store/tests/rekey-reason/rekey-reason.actions.ts]: weil do

File: src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts [hunk 6]
```diff
@@ -217,7 +220,19 @@ export abstract class WaitingRoomToCarBasePageComponent extends BasePageComponen
...
+
+  async presentAlert() {
```
Comment (matthew2564) [inline src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts]: Can we just give this a name thats more specific to test-centre-journal please?

File: src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts (hunk 5 above)
Comment (andrewsetterfield) [reply src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts]: its temporary code until a back log story is delivered but agree, will ammend to the standard

File: src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts (hunk 6 above)
Comment (andrewsetterfield) [reply src/app/shared/classes/test-flow-base-pages/waiting-room-to-car/waiting-room-to-car-base-page.ts]: sure

File: src/app/pages/test-report/components/toolbar/toolbar.scss (hunk 1 above)
Comment (andrewsetterfield) [reply src/app/pages/test-report/components/toolbar/toolbar.scss]: already commented that i'd remove that :D

File: src/store/tests/rekey-reason/rekey-reason.actions.ts (hunk 3 above)
Comment (andrewsetterfield) [reply src/store/tests/rekey-reason/rekey-reason.actions.ts]: reverted. whoops.

**Review by solomonsimukoko — APPROVED** [review]

**Review by matthew2564 — APPROVED** [review]

**Pull request merged**